src/NUnitFramework/framework/Attributes/ValuesAttribute.cs

```diff
@@ -101,7 +103,51 @@ public ValuesAttribute(params object[] args)
         /// <param name="parameter">The parameter of a parameterized test.</param>
         public IEnumerable GetData(Type fixtureType, ParameterInfo parameter)
         {
+            if(data.Length == 0)           
+                return GenerateData(parameter.ParameterType);
+            else           
             return ParamAttributeTypeConversions.ConvertData(data, parameter.ParameterType);
         }
+
+        /// <summary>
+        /// To generate data for Values attribute, in case no data is provided.
+        /// </summary>
+        /// <param name="targetType"></param>
+        /// <returns></returns>
+        private static IEnumerable GenerateData(Type targetType)
+        {
+            if (IsNullableEnum(targetType))
+            {
+                var enumValues = Enum.GetValues(Nullable.GetUnderlyingType(targetType));
+                var enumValuesWithNull = new object[enumValues.Length + 1];
+                Array.Copy(enumValues, enumValuesWithNull, enumValues.Length);
+                return enumValuesWithNull;
+            }
+            if (targetType.GetTypeInfo().IsEnum)
+            {
+                return Enum.GetValues(targetType);
+            }
+            if (targetType == typeof(bool?))
+            {
+                return new object[] { null, true, false };
+            }
+            if (targetType == typeof(bool))
+            {
+                return new object[] { true, false };
+            }
+
+            return new object[] { };
+        }      
+
+        /// <summary>
+        /// To Check if type is nullable enum.
+        /// </summary>
+        /// <param name="t"></param>
+        /// <returns></returns>
+        private static bool IsNullableEnum(Type t)
```

Comment:
Personally, I think that the `param` and `returns` are unnecessary if they don't contain any information.